x/concentrated-liquidity/fees_test.go

```diff
@@ -74,41 +74,44 @@ func (s *KeeperTestSuite) TestInitializeFeeAccumulatorPosition() {
 
 func (s *KeeperTestSuite) TestGetFeeGrowthOutside() {
 	type feeGrowthOutsideTest struct {
-		poolSetup           bool
-		tickSetup           bool
-		feeAccumulatorSetup bool
-		expectedError       bool
+		poolSetup                bool
+		tickSetup                bool
+		expectedFeeGrowthOutside sdk.DecCoins
+		invalidTick              bool
+		expectedError            bool
 	}
 
+	defaultAccumCoins := sdk.NewDecCoins(sdk.NewDecCoin("foo", sdk.NewInt(50)))
+	defaultPoolId := uint64(1)
+	defaultLiquidityForLowerTick := sdk.MustNewDecFromStr("10.0")
+	defaultLiquidityForUpperTick := sdk.MustNewDecFromStr("20.0")
+
+	defaultUpperTickIndex := int64(5)
+	defaultLowerTickIndex := int64(3)
+
 	tests := map[string]feeGrowthOutsideTest{
-		// TODO: uncomment this once tickInfo feeGrowthOutside logic has been implemented
-		"happy path": {
-			poolSetup:           true,
-			tickSetup:           true,
-			feeAccumulatorSetup: true,
-			expectedError:       false,
+		"fee growth outside with tick initialized": {
+			poolSetup:                true,
+			tickSetup:                true,
+			expectedFeeGrowthOutside: defaultAccumCoins,
```

Comment:
We need to have more tests for various cases of what fee growth outside is.

For example, assume we have a position at ticks 10 and 20. As a swap is completed, we cross the ticks from left to right. Both ticks are crossed and the current tick stops at tick 25. As a result, both tick 10 and tick 20 are updated with the `feeGrowthGlobal`.

We need to simulate and test all the edge cases such as:
- crossing left to right
   *  the current price stays before the lower tick so none of the tick accumulators are updated
   *  the current price stays in-between the lower and upper tick so only lower tick is updated
   *  the current price stays after the upper tick so both lower and upper tick accumulators are updated
- crossing right to left
   * similar test cases to left to right but in the other direction.


I did a similar test for `collectFees` in https://github.com/osmosis-labs/osmosis/pull/3893
Test link:
https://github.com/osmosis-labs/osmosis/blob/e1e8a4624fdd844fdd8536de13ca8924a3c51de2/x/concentrated-liquidity/fees_test.go#L400

We need something similar but more focused on `feeGrowthOutside` specifically. I think it's important that we test all the edge cases at a lower level of abstraction. Otherwise, if we miss something, it will be close to impossible to debug at a higher level of abstraction (e.g. in swaps) so we will have to go back here.